### PR TITLE
test: additional slashed operator tests

### PR DESCRIPTION
**Pull request by 0xClandestine**
**Motivation:**

Need tests to ensure proper handling of edge cases when stakers interact with operators who have been fully slashed.

**Modifications:**

- Added new integration test file `SlashedOperator.t.sol` with three key test cases:
  - [x] `testFuzz_register_allocate_fullSlash_deposit_delegate`: Tests that delegating to a fully slashed operator fails after deposits
  - [x] `testFuzz_register_allocate_fullSlash_delegate_deposit`: Tests the interaction when delegation to a slashed operator occurs before deposits
  - [x] `testFuzz_register_allocate_fullSlash_delegate_redelegate_deposit`: Tests the redelegation flow when a staker moves from a slashed operator to a new operator

**Result:**

After these changes, the protocol has better test coverage for slashed operator scenarios.

### CODE REVIEW

File: src/test/integration/tests/SlashedOperator.t.sol [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+    }
+
+    function testFuzz_register_allocate_fullSlash_deposit_delegate(uint24 r) public rand(r) {
+        // 5) Staker deposits into strategies.
+        staker.depositIntoEigenlayer(strategies, initTokenBalances);
+        initDepositShares = _calculateExpectedShares(strategies, initTokenBalances);
+        check_Deposit_State(staker, strategies, initDepositShares);
+
+        // 6) Staker delegates to operator who is fully slashed, should fail.
+        vm.expectRevert();
```
Comment (ypatil12) [inline src/test/integration/tests/SlashedOperator.t.sol:57]: Nit: add revert message

File: src/test/integration/tests/SlashedOperator.t.sol [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+    }
+
+    function testFuzz_register_allocate_fullSlash_delegate_deposit(uint24 r) public rand(r) {
+        // 5) Staker delegates to operator who is fully slashed, should fail.
+        staker.delegateTo(operator);
+        // NOTE: Leads to division by zero?
+        // check_Delegation_State(staker, operator, strategies, new uint256[](strategies.length)); // Initial shares are zero
+        
+        // 6) Staker deposits into strategies, should fail.
+        vm.expectRevert();
```
Comment (ypatil12) [inline src/test/integration/tests/SlashedOperator.t.sol:68]: Nit: add revert

File: src/test/integration/tests/SlashedOperator.t.sol [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+
+        User newOperator = _newRandomOperator_NoAssets();
+        newOperator.registerForOperatorSet(operatorSet);
+
+        // 6) Staker redelegates to new operator.
+        Withdrawal[] memory withdrawals = staker.redelegate(newOperator);
+        bytes32[] memory withdrawalRoots = _getWithdrawalHashes(withdrawals);
+        check_Redelegate_State(staker, operator, newOperator, withdrawals, withdrawalRoots, strategies, new uint256[](strategies.length));
+
+        _rollBlocksForCompleteWithdrawals(withdrawals);
```
Comment (ypatil12) [inline src/test/integration/tests/SlashedOperator.t.sol:84]: Can we complete as shares for sanity, I know we missed that in the test description

File: src/test/integration/tests/SlashedOperator.t.sol [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+        // 6) Staker delegates to operator who is fully slashed, should fail.
+        vm.expectRevert();
+        staker.delegateTo(operator);
+    }
+
+    function testFuzz_register_allocate_fullSlash_delegate_deposit(uint24 r) public rand(r) {
+        // 5) Staker delegates to operator who is fully slashed, should fail.
+        staker.delegateTo(operator);
+        // NOTE: Leads to division by zero?
+        // check_Delegation_State(staker, operator, strategies, new uint256[](strategies.length)); // Initial shares are zero
```
Comment (ypatil12) [inline src/test/integration/tests/SlashedOperator.t.sol:65]: Interesting, I assume this is from the `_calcWithdrawableShares`? Perhaps we can just assert that the staker is delegated via a manual check, and then add a comment why we don't use the big `check_Delegation_state`

File: src/test/integration/tests/SlashedOperator.t.sol [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+
+        // 6) Staker redelegates to new operator.
+        Withdrawal[] memory withdrawals = staker.redelegate(newOperator);
+        bytes32[] memory withdrawalRoots = _getWithdrawalHashes(withdrawals);
+        check_Redelegate_State(staker, operator, newOperator, withdrawals, withdrawalRoots, strategies, new uint256[](strategies.length));
+
+        _rollBlocksForCompleteWithdrawals(withdrawals);
+
+        // 7) Staker deposits into strategies.
+        staker.depositIntoEigenlayer(strategies, initTokenBalances);
```
Comment (ypatil12) [inline src/test/integration/tests/SlashedOperator.t.sol:87]: Staker can deposit prior to rolling blocks for complete withdrawal (right after redelegating)

Comment (0xClandestine) [reply src/test/integration/tests/SlashedOperator.t.sol:85]: [1dd8033](https://github.com/Layr-Labs/eigenlayer-contracts/pull/1198/commits/1dd8033bef72857b136ce8b1dacec43123c162d4)

File: src/test/integration/tests/SlashedOperator.t.sol (hunk 1 above)
Comment (0xClandestine) [reply src/test/integration/tests/SlashedOperator.t.sol]: [0d9aa7d](https://github.com/Layr-Labs/eigenlayer-contracts/pull/1198/commits/0d9aa7db588aea5ef74d9a8b1fcb8abbb1aa4a15)

File: src/test/integration/tests/SlashedOperator.t.sol (hunk 4 above)
Comment (0xClandestine) [reply src/test/integration/tests/SlashedOperator.t.sol]: cdbe9e2

File: src/test/integration/tests/SlashedOperator.t.sol (hunk 3 above)
Comment (0xClandestine) [reply src/test/integration/tests/SlashedOperator.t.sol]: [a27a176](https://github.com/Layr-Labs/eigenlayer-contracts/pull/1198/commits/a27a176930dba62a3171fa4434ced5ca0e221a49)

**Review by ypatil12 — APPROVED** [review]
LGTM :)
